crates/engine/tree/src/tree/mod.rs

```diff
@@ -1348,7 +1373,13 @@ where
     /// Returns a batch of consecutive canonical blocks to persist in the range
     /// `(last_persisted_number .. canonical_head - threshold]` . The expected
     /// order is oldest -> newest.
-    fn get_canonical_blocks_to_persist(&self) -> Vec<ExecutedBlockWithTrieUpdates<N>> {
+    fn get_canonical_blocks_to_persist(
```

Comment:
can we add a note here about when this would return an error?

Reply:
added